app/src/main/java/com/readrops/app/more/preferences/PreferencesScreenModel.kt

```diff
@@ -1,21 +1,41 @@
 package com.readrops.app.more.preferences
 
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.produceState
+import androidx.compose.ui.res.stringResource
 import cafe.adriel.voyager.core.model.StateScreenModel
 import cafe.adriel.voyager.core.model.screenModelScope
+import com.readrops.app.R
 import com.readrops.app.util.Preference
 import com.readrops.app.util.Preferences
+import com.readrops.db.Database
+import com.readrops.db.entities.Item
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.get
+import org.koin.core.component.inject
 
 typealias PreferenceState<T> = Pair<T, Preference<T>>
 
 class PreferencesScreenModel(
     preferences: Preferences,
     dispatcher: CoroutineDispatcher = Dispatchers.IO
-) : StateScreenModel<PreferencesScreenState>(PreferencesScreenState.Loading) {
+) : StateScreenModel<PreferencesScreenState>(PreferencesScreenState.Loading), KoinComponent {
+    val exampleItem = flow<Item> {
+        val database = get<Database>()
+        if(database.itemDao().count() > 0) {
+            emit(database.itemDao().selectFirst())
+        }
+    }
+
+
+    val showDialog = mutableStateOf(false)
```

Comment:
Oh okay, thank you for this. When I talked about putting this kind of state in the screenModel, I didn't mention that it should go in `PreferencesScreenState.Loaded`. Of course, to update it, you will need a method and cast `mutableState` to `PreferencesScreenState.Loaded` to be able to change `showDialog`. I do this in all the other screens, you could have a look.

Reply:
I'm sorry, I'm not sure I understand what you mean here. Are you talking about:

```suggestion
    private val _showDialog = mutableStateOf(false)
    var showDialog
        get() = _showDialog.value
        set(value) {
            _showDialog.value = value
        }
```

Reply:
More something like this:

```kotlin
    sealed class PreferencesScreenState {
    data object Loading : PreferencesScreenState()
    data object Error : PreferencesScreenState()

    data class Loaded(
        val themePref: PreferenceState<String>,
        val backgroundSyncPref: PreferenceState<String>,
        val scrollReadPref: PreferenceState<Boolean>,
        val hideReadFeeds: PreferenceState<Boolean>,
        val openLinksWith: PreferenceState<String>,
        val timelineItemSize: PreferenceState<String>,
        val mainFilterPref: PreferenceState<String>,
        val syncAtLaunchPref: PreferenceState<Boolean>,
        val showDialog: Boolean = false
    ) : PreferencesScreenState()
    
    fun updateDialog(isVisible: Boolean) {
        if (mutableState.value is PreferencesScreenState.Loaded) {
            mutableState.update { (mutableState.value as PreferencesScreenState.Loaded).copy(showDialog = isVisible) }
        }
    }

}
```